test/opt/fold_test.cpp

```diff
@@ -799,7 +799,15 @@ ::testing::Values(
             "%2 = OpFDiv %float %float_n1 %float_0\n" +
             "OpReturn\n" +
             "OpFunctionEnd",
-        2, -std::numeric_limits<float>::infinity())
+        2, -std::numeric_limits<float>::infinity()),
+    // Test case 3: Fold -2.0
+    InstructionFoldingCase<float>(
+        Header() + "%main = OpFunction %void None %void_func\n" +
+            "%main_lab = OpLabel\n" +
+            "%2 = OpFNegate %float %float_2\n" +
+            "OpReturn\n" +
+            "OpFunctionEnd",
+        2, -2)
```

Comment:
I know we don't do this for many cases, but could you add a test for when the input is the result of and OpConstantNull.

Also add a test for a vector.

Reply:
Test double as well.

Reply:
Done